envi/memory.py

```diff
@@ -525,31 +525,78 @@ def getMemoryMap(self, va):
     def getMemoryMaps(self):
         return [mmap for mva, mmaxva, mmap, mbytes in self._map_defs]
 
-    def readMemory(self, va, size):
+    def readMemory(self, va, size, origva=None):
```

Comment:
I don't love that the internal `origva` parameter here is exposed to any user. There's no indication if this is a parameter is something that a user should care about or provide. Some options:

  - document that `origva` is only to be used internally
  - refactor this routine to avoid recursion
  - refactor the routine to an inner `_readMemory(..., origva)` and outer public `readMemory(va, size)`.

Reply:
i want the third option as well, and possibly even the second.  

since this is at the core of every single emulation thing, i'm hesitant to add the additional call even.  

this current approach was taken because it only did a small number comparison, and only really adds cost if we cross memory boundaries.

for now, i'll comment the origva as "internal use only"

Reply:
updated the code docstring and changed the name of the arg to `_origva` to further discourage use